implementations/rust/ockam/ockam_entity/src/channel.rs

```diff
@@ -11,32 +11,51 @@ pub(crate) use listener::*;
 mod messages;
 pub(crate) use messages::*;
 
-impl<V: ProfileVault> ProfileImpl<V> {
+#[allow(dead_code)]
+struct SecureChannel {}
+
+impl SecureChannel {
     /// Create mutually authenticated secure channel
-    pub async fn create_secure_channel<A: Into<Route>>(
-        &mut self,
+    #[allow(dead_code)]
+    pub async fn create_secure_channel<R, V, P>(
         ctx: &Context,
-        route: A,
-    ) -> Result<Address> {
-        Initiator::create(ctx, route, self).await
+        profile: &mut P,
+        route: R,
+    ) -> Result<Address>
+    where
+        R: Into<Route>,
+        V: ProfileVault,
+        P: ProfileTrait<V>,
+    {
+        Initiator::create(ctx, route, profile).await
     }
 
     /// Create mutually authenticated secure channel listener
-    pub async fn create_secure_channel_listener<A: Into<Address>>(
-        &mut self,
```

Comment:
What is the reason why it's no longer a Profile method?

Reply:
Because it can't be expressed as a trait. I tried using async trait but there were some other constraints in the signature.

Reply:
This is functionality that Entity needs to proxy to Profile somehow, so we either need to refactor the signature to be friendlier to a trait that both can implement, or manually add the methods on Entity.

Reply:
Interesting, don't see anything obvious why this can't be a trait

Reply:
Let's pair after demos tomorrow, we can go through and see where I got stuck